server/swap/swap.go

```diff
@@ -1525,18 +1589,20 @@ func (s *Swapper) processInit(msg *msgjson.Message, params *msgjson.Init, stepIn
 		log.Errorf("Contract txn located after match was revoked (match id=%v, maker=%v)",
 			matchID, actor.isMaker)
 		s.respondError(msg.ID, actor.user, msgjson.ContractError, "match already revoked due to inaction")
-		return wait.DontTryAgain
+		return dontTryAgain()
 	}
 
 	actor.status.mtx.Lock()
-	actor.status.swap = contract
+	actor.status.swap = contract // swap should not be set already (handleInit should gate)
```

Comment:
If we were not confident that the searching semaphore is sufficient to prevent this as well as simultaneous coin waiters, we could add a final nil check here before assigning, but I don't think we need to code for every conceivable bug.  Besides, I'm not sure what an appropriate way to handle that would be other than log an error or panic.